MantidQt/CustomInterfaces/src/DynamicPDF/SliceSelector.cpp

```diff
@@ -129,14 +173,101 @@ void SliceSelector::launchBackgroundRemover() {
   //}
   // m_BackgroundRemover->refreshSlice(m_loadedWorkspace,
   // m_selectedWorkspaceIndex);
-  g_log.error("Not implemented...yet");
+  auto title = QString::fromStdString(this->name());
+  auto error = QString::fromStdString("Not so fast, cowboy! (not implemented)");
+  QMessageBox::warning(this, title, error);
 }
 
-/// Opens the Qt help page for the interface
+/**
+ * @brief Opens the Qt help page for the interface
+ */
 void SliceSelector::showHelp() {
   MantidQt::API::HelpWindow::showCustomInterface(
       NULL, QString("Dynamic PDF Calculator"));
 }
+
+
+/*        ***********************
+ *        **  Private Members  **
+ *        ***********************/
+
+
+/**
+ * @brief Initialize UI form, spawn picker line, connect SIGNALS/SLOTS
+ */
+void SliceSelector::initLayout() {
+  m_uiForm.setupUi(this);
+  this->spawnPickerLine();
+  connect(m_uiForm.dataSelector, SIGNAL(dataReady(const QString &)), this,
+          SLOT(loadSlices(const QString &)));
+  connect(m_uiForm.pushHelp, SIGNAL(clicked()), this, SLOT(showHelp()));
+  connect(m_uiForm.spinboxSliceSelector, SIGNAL(valueChanged(int)), this,
```

Comment:
`valueChanged(int)` this seems to update the plot every time the spin box is changed, so when I'm trying to type `102` in the `spinBox` it will plot `1` then `10` and then `102`.. As it is taking so long to plot the graph; maybe a better option would be to use `update button` which updates the plot when user finish typing or dragging up and down.